src/eredis_cluster_manager.erl

```diff
@@ -0,0 +1,415 @@
+get_key_from_command([[X|Y]|Z]) when is_list(X) ->
+    case string:to_lower(X) of
+        "multi" ->
+            get_key_from_command(Z);
+        _ ->
+            get_key_from_command([X|Y])
+    end;
+get_key_from_command([Term1,Term2|Rest]) when is_bitstring(Term1) ->
+    get_key_from_command([bitstring_to_list(Term1),Term2|Rest]);
+get_key_from_command([Term1,Term2|Rest]) when is_bitstring(Term2) ->
+    get_key_from_command([Term1,bitstring_to_list(Term2)|Rest]);
+get_key_from_command([Term1,Term2|Rest]) ->
+    case string:to_lower(Term1) of
+        "info" ->
+            undefined;
+        "config" ->
+            undefined;
+        "shutdown" ->
+            undefined;
+        "slaveof" ->
+            undefined;
+        "eval" ->
+            get_key_from_rest(Rest);
+        "keys" ->
+            keys;
+        "evalsha" ->
+            get_key_from_rest(Rest);
+        _ ->
+            Term2
+    end;
+get_key_from_command(_) ->
+    undefined.
+
+%% =============================================================================
+%% @doc Get key for command where the key is in th 4th position (eval and
+%% evalsha commands)
+%% @end
+%% =============================================================================
+-spec get_key_from_rest([anystring()]) -> string() | undefined.
+get_key_from_rest([_,KeyName|_]) when is_bitstring(KeyName) ->
+    bitstring_to_list(KeyName);
+get_key_from_rest([_,KeyName|_]) when is_list(KeyName) ->
+    KeyName;
+get_key_from_rest(_) ->
+    undefined.
+
+%% =============================================================================
+%% @doc Return the hash slot from the key
+%% @end
+%% =============================================================================
+-spec get_key_slot(Key::any()) -> Slot::integer().
+get_key_slot(K) ->
+    %% cast any type of key to a list.
+    Key = lists:concat([K]),
+    KeyToBeHashed = case string:chr(Key,${) of
+                        0 ->
+                            Key;
+                        Start ->
+                            case string:chr(string:substr(Key,Start+1),$}) of
+                                0 ->
+                                    Key;
+                                Length ->
+                                    if
+                                        Length =:= 1 ->
+                                            Key;
+                                        true ->
+                                            string:substr(Key,Start+1,Length-1)
+                                    end
+                            end
+                    end,
+    eredis_cluster_hash:hash(KeyToBeHashed).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Methods to manage list of eredis connections
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+init_eredis_connections() ->
+    ets:new(?MODULE, [protected, set, named_table, {read_concurrency, true}]).
```

Comment:
this means you are sharing this table between cluster. I don't think we want this. We should have ets table per cluster.